Remove usage of global options from iptables cell

pippolo84 opened the pull request:
Remove usage of `option.Config` in iptables cell. These are just leftover after the cell has been modularized.

The option `prepend-iptables-chains` used to have an env var alias named `CILIUM_PREPEND_IPTABLES_CHAIN`.  It was there to ensure backward compatibility.  ~But the alias is exactly the same string as the usual name for the option (prefix is "CILIUM_" and option name is "PREPEND_IPTABLES_CHAIN"). In other words, there was actually no alias.
Thus it is safe to move the option in the cell.Config(), even if hive package does not support additional aliasies for the same config option.~
`CILIUM_PREPEND_IPTABLES_CHAIN` alias has been deprecated with a warning and a note in the v1.15 Upgrade Notes. Read [this comment](https://github.com/cilium/cilium/pull/29088#issuecomment-1812314672) for more info.

~Depends on: https://github.com/cilium/cilium/pull/28746~

pippolo84 committed:
iptables: Remove usage of global option EnableIPv6

Since iptables manager has been modularized, it should not depend
directly on the global daemon config. Instead, it should read the
configuration from its cell.

Signed-off-by: Fabio Falzoi <fabio.falzoi@isovalent.com>

pippolo84 committed:
iptables: Move prepend-iptables-chains option into the cell

prepend-iptables-chains is used to enable prepending iptables chains
instead of appending, and it is relevant only for the iptables manager.
Since the iptables manager has been modularized, let's move this into
the cell "private" config.

The option used to have an env var alias named
`CILIUM_PREPEND_IPTABLES_CHAIN`, for backward compatibility reason. The
alias has been deprecated and a warning will be printed in case that env
var will be found not empty at iptables cell startup.
Besides, the env var deprecation has been described in the v1.15 upgrade
notes.

Signed-off-by: Fabio Falzoi <fabio.falzoi@isovalent.com>

pippolo84 committed:
iptables: Use cell config in iptables tests

iptables manager has been modularized into a cell, so the tests should
config it through the fields in the manager struct itself, not through
the global config options.

Signed-off-by: Fabio Falzoi <fabio.falzoi@isovalent.com>

thorn3r reviewed (commented):
changes look fairly straight-forward, but looks like there's quite a few tests failing

pippolo84 committed:
iptables: Copy config at cell startup

Copy the cell specific config parameters passed through dependency
injection into the manager struct.

Fixes: a6a2b73f4f ("iptables: Add a cell for iptables config manager")

Signed-off-by: Fabio Falzoi <fabio.falzoi@isovalent.com>

asauber reviewed (commented):
Re-ran failing clustermesh test

asauber reviewed (commented):
Hey @pippolo84, would you be able to link to some documentation (or code) that indicates that the env var alias would be preserved?

 Alternatively, can you provide some interactive output when running the agent which shows that it is used with the new implementation?

learnitall commented on Documentation/operations/upgrade.rst:
```diff
@@ -339,6 +339,11 @@ Annotations:
+  cannot be used anymore to enable or disable Cilium iptables rules prepend instead
+  of append.
+  Please use the ``CILIUM_PREPEND_IPTABLES_CHAINS`` environment variable (note the
+  trailing ``S``) or the command line flag ``--prepend-iptables-chains`` instead.
```
Can we change up the wording here to include a bit of information regarding why this change was made? Maybe something like:

```suggestion
* The ``CILIUM_PREPEND_IPTABLES_CHAIN`` environment variable has been renamed
  to ``CILIUM_PREPEND_IPTABLES_CHAINS`` (note the trailing ``S``) to more accurately
  match the name of the associated command line flag ``--prepend-iptables-chains``.
```

pippolo84 replied:
Fixed, thanks!

learnitall reviewed (changes requested):
Small nit on the upgrade notes, but otherwise LGTM.

thorn3r reviewed (approved):
aside from the open conversation, LGTM!

learnitall reviewed (approved):
Looks good for docs, thanks!

ldelossa reviewed (approved):
Sorry for delay, LGTM :)
